osm_cli/cli/main.py

```diff
@@ -13,6 +14,30 @@ def osm():
 @click.argument("file_path", type=click.Path(exists=True))
 @click.argument("file_id", type=str)
 def pdf_xml(file_path, file_id):
-    """This function converts a file from PDF to XML"""
-    # Function Implementation
+    """
+    Args:
+        file_path (file path): First parameter
+        file_id (string): Second parameter
+    Returns:
+        Creates an XML file in the directory xmls_sciencebeam
+    """
+    try:
+        converter = PDFConverter()
+
+        if not converter.is_docker_running():
+            raise click.ClickException('Please make sure the docker is running')
+
+        if not converter.is_host_ready():
+            raise click.ClickException('The converter server is offline')
+
+        xml_content = converter.convert(file_path)
+        # Save the converted xml contents
+        output_file: str = f'docs/examples/sciencebeam_xml_outputs/{file_id}.xml'
+        with open(output_file, 'w', encoding='utf-8') as xml_file:
```

Comment:
Use pathlib.Path throughout the codebase. It is a more concise and less error prone way to work with files. In this case

```
Path(xml_file).write_text(xml_content)
```

But xml_file should already be a Path type in which case you shouldn't needlessly coerce so just

```
xml_file.write_text(xml_content)
```

Reply:
We have fixed it by updating the code to use pathlib.Path throughout the codebase.